File: CoreTestDataConverter/Converter/StatusCalculator/StatusCalculatorTestDataConverter.cs
using CoreTestDataConverter.Model.StatusCalculator;
using SaveData.Model.Development;

namespace CoreTestDataConverter.Converter.StatusCalculator
{
	public static class StatusCalculatorTestDataConverter
	{
		public static StatusCalculatorTestData ToTestData(CalculationTestSource testSource)
		{
			if (testSource == null)
			{
				return null;
			}

			return new StatusCalculatorTestData
			{
				Input = new StatusCalculatorInputTestData
				{
					BaseStatus = StatusConverter.ToTestData(testSource.BaseStatus),
					MitamaStatus = StatusConverter.ToTestData(testSource.MitamaStatus)
				},
				Expected = StatusConverter.ToTestData(testSource.FinalStatus)
			};
		}
	}
}

File: CoreTest/TestData/Converter/StatusCalculator/StatusCalculatorTestDataConverter.cpp
#include "pch.h"
#include "StatusCalculatorTestDataConverter.h"

#include "../StatusConverter.h"
#include "StatusCalculatorInputConverter.h"

StatusCalculatorTestData StatusCalculatorTestDataConverter::toTestData(const nlohmann::json& json)
{
	StatusCalculatorTestData data{};

	data.Input = StatusCalculatorInputConverter::toTestData(json.at("Input"));
	data.Expected = StatusConverter::toTestData(json.at("Expected"));

	return data;
}
